[Gibson0918] IP

Gibson0918 opened the pull request:
## Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke is a simple reminder app that helps you to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ *SUPER FAST* to use

All you need to do is, 
1. download it from [here](https://github.com/Gibson0918/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😁

And it is **FREE**!

Features:
- [x] Adding ToDo task
- [x] Adding task with deadline
- [x] Adding Event with deadline

if you are a Java programmer, you can use it to practice Java too. Here's the main method:
```public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Gibson0918 committed:
Add Level 1. Greet, Echo, Exit feature

Gibson0918 committed:
Level 2 Completed

Gibson0918 committed:
Level 3 done

Gibson0918 committed:
level 4 completed

Gibson0918 committed:
Update Duke.java

Gibson0918 committed:
Completed Duke's Automated Text UI testing

Gibson0918 committed:
Level 5 Completed.

Gibson0918 committed:
Level 6 Completed

Gibson0918 committed:
Added Enumerations

Gibson0918 committed:
Update Duke.java

Gibson0918 committed:
Delete unnecessary files

Gibson0918 committed:
Level 7 Completed

Gibson0918 committed:
Level 8 Completed

Gibson0918 committed:
Merge branch 'branch-Level-8'

Guo-KeCheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+        String[] userInput = sc.nextLine().split(" ", 2);
+    
+        try {
+            switch (userInput[0]) {
+                case "bye":
+                    System.out.println("Bye. Hope to see you again soon!");
+                    sc.close();
+                    startDuke = false;
+                    break;
```
Switch-case should not be indented. Could edit coding style in Intellij to reduce having to manually check.

Guo-KeCheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+        String[] userInput = sc.nextLine().split(" ", 2);
+    
+        try {
+            switch (userInput[0]) {
```
Could potentially abstract out userInput[0] using enum

Guo-KeCheng reviewed (commented):
LGTM. Good job thus far. Jiayous.

Gibson0918 committed:
Refactor Duke to use more OOP

Gibson0918 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Gibson0918 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Gibson0918 committed:
Fixed bugs related to datetime and organize duke into packages

Gibson0918 committed:
add Gradle to Duke

sarthak181 commented on src/main/java/Event.java:
```diff
@@ -1,14 +1,18 @@
-    protected String from;
-    protected String to;
+    protected LocalDateTime from;
+    protected LocalDateTime to;
```
perhaps startDate and endDate would be better names

sarthak181 commented on src/main/java/Event.java:
```diff
@@ -1,14 +1,18 @@
     public String toString() {
-        return "[E]" + super.toString() + " (from: " + from + " to: " + to + ")";
+        return "[E]" + super.toString() + " (from: " + from.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + " to: " + to.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + ")";
+
```
this return statement could be indented better

sarthak181 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,21 @@
+public abstract class Command {
+    private boolean isTerminated = false;
+
+    public boolean isTerminated() {
+        return isTerminated;
+    };
```
there shouldn't be a semi colon at the end of a method

sarthak181 reviewed (commented):
Great code overall!

Gibson0918 committed:
Added JUnit tests for Duke

Gibson0918 committed:
Merge branch 'master' of https://github.com/Gibson0918/ip

Gibson0918 committed:
Release duke to the wild

rachtan27 commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,16 @@
+public class ExitCommand extends Command {
+
+    @Override
+    public void initCommand(TaskList tasks, Ui ui, Storage storage) {
```
Good job making the code readable and easy to understand

rachtan27 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,59 @@
+        String[] output = command.split(" ", 2);
+        ValidCommands validCommands = ValidCommands.valueOf(output[0].toUpperCase());
+        switch (validCommands) {
+            case LIST:
```
Remember that there should not be any indentation for case clauses :)

rachtan27 replied:
otherwise, i like how it is very easy to understand what your parser is doing 👍

rachtan27 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,82 @@
+                String[] data = sc.nextLine().split(" \\| ");
+                Task task = null;
+                switch (data[0]) {
+                    case "T":
```
Same issue with the indentation for switch case as mentioned earlier :)

rachtan27 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,82 @@
+    public List<Task> loadFile() throws FileNotFoundException {
+        List<Task> taskList = new ArrayList<>();
+        File file = new File(filePath);
+        if (file.isFile()) {
```
Is it possible to avoid this deep nesting?

Gibson0918 committed:
Added Java Docs to Duke

Gibson0918 committed:
Added StringBuilder class to Duke

Gibson0918 committed:
Completed Level 9

Gibson0918 committed:
Tweak Duke's code

Gibson0918 committed:
Level 10 Completed

Gibson0918 committed:
Remove commented lines

Gibson0918 committed:
Added more Java Docs

Gibson0918 committed:
Improve Duke's code quality

Remove Java Docs for methods that have been removed.

Refactor Todo, Deadline and Event's toString() and formatForFile() method for better clarity.

Removed unused method in the Ui Class

Gibson0918 committed:
Update README.md

Gibson0918 committed:
Add Update feature to Duke

Tasks already added to Duke cannot be updated so an edit/update feature for existing task would be appreciated as the only way to modify a task would be to delete the existing one and create a new one.

Let's modify the 3 tasks (Event, Todo, Deadline) and add an UpdateCommand class to support Duke in updating existing task

Gibson0918 committed:
Updated GUI and added set method to tasklist

1. Resolve minor GUI bugs where the GUI was not resizing properly

2. Added the missed out set method in TaskList class

Gibson0918 committed:
Added Assertions to Duke

Gibson0918 committed:
Merge pull request #2 from Gibson0918/branch-A-Assertions

Added Assertions to Duke

Gibson0918 committed:
Update README.md

Gibson0918 committed:
Fixed some bugs relating to DateTime format and Duke exit command

Gibson0918 committed:
Merge branch 'master' of https://github.com/Gibson0918/ip

Gibson0918 committed:
Added Ui image for docs

Gibson0918 committed:
Clean up code for checkstyle
